x-pack/distributions/internal/observabilitySRE/qa/smoke/docker/logstash/pipeline/logstash.conf

```
@@ -0,0 +1,71 @@
+input {
+  beats {
+    port => 5044
+    ssl_enabled => true
+    ssl_certificate_authorities => ["/usr/share/logstash/config/certs/ca.crt"]
+    ssl_certificate => "/usr/share/logstash/config/certs/logstash.crt"
+    ssl_key => "/usr/share/logstash/config/certs/logstash.key"
+    ssl_client_authentication => "required"
+    ssl_supported_protocols => ["TLSv1.2"]
+  }
+}
+
+filter {
+  grok {
+    match => { "message" => "TEST-LOG: %{GREEDYDATA:log_content}" }
+  }
+  
+  if [log_content] =~ /timestamp=/ {
+    grok {
+      match => { "log_content" => ".*timestamp=%{TIMESTAMP_ISO8601:timestamp}.*" }
+    }
+    date {
+      match => [ "timestamp", "ISO8601" ]
+      target => "@timestamp"
+    }
+  }
+
+  age {}
+  if [@metadata][age] > 86400 {
+    mutate {
+      add_tag => ["old_event"]
+    }
+  }
+  
+  if [log_content] =~ /DEBUG/ {
+    drop { }
+  }
+  
+  if [log_content] =~ /json=/ {
+    grok {
+      match => { "log_content" => ".*json=%{GREEDYDATA:json_string}.*" }
```

since grok pattern is unanchored-by-default, we don't need the leading and trailing `.*`'s.

```suggestion
      match => { "log_content" => "json=%{GREEDYDATA:json_string}" }
```

If we want to safeguard for the json to start with `{` and end with `}`, we could add lookarounds:

```suggestion
      match => { "log_content" => "json=(?={)%{GREEDYDATA:json_string}(?<=})" }
```

---

Thanks! Given we have full control of the input file i think keeping this as simple as possible would be ideal. I went with the removal of the wildcard matchers for simplicity.